fpsd/features.py

```diff
@@ -0,0 +1,765 @@
+        11         | 652      | 77       | 729
+
+        where a, b, c are the feature columns:
+
+        total_number_of_incoming_cells,
+        total_number_of_outgoing_cells,
+        and total_number_of_cells.
+
+        Returns:
+            [string] name of newly created table
+        """
+
+        self.drop_stale_feature_table("features.cell_numbers")
+
+        query = ("CREATE TABLE features.cell_numbers AS            "
+                 "(SELECT t1.exampleid, t1.total_number_cells,     "
+                 " t2.total_number_incoming_cells,                 "
+                 " t3.total_number_outgoing_cells FROM             "
+                 "   (SELECT exampleid, count(*)                   "
+                 "    AS total_number_cells                        "
+                 "    FROM raw.frontpage_traces                    "
+                 "    GROUP BY exampleid) t1                       "
+                 " LEFT OUTER JOIN                                 "
+                 "   (SELECT exampleid, count(*)                   "
+                 "    AS total_number_incoming_cells               "
+                 "    FROM raw.frontpage_traces                    "
+                 "    WHERE raw.frontpage_traces.ingoing = 't'     "
+                 "    GROUP BY exampleid) t2                       "
+                 " ON t1.exampleid = t2.exampleid                  "
+                 " LEFT OUTER JOIN                                 "
+                 "   (SELECT exampleid, count(*)                   "
+                 "    AS total_number_outgoing_cells               "
+                 "    FROM raw.frontpage_traces                    "
+                 "    WHERE raw.frontpage_traces.ingoing = 'f'     "
+                 "    GROUP BY exampleid) t3                       "
+                 " ON t1.exampleid = t3.exampleid);                ")
+        self.execute_query_from_string(query)
+
+        return "features.cell_numbers"
+
+    def generate_table_cell_timings(self):
+        """This method takes all examples and produces a
+        table features.cell_timings with the following format:
+
+        exampleid  | total_elapsed_time
+        (integer)  | (numeric)
+        -----------------------------------------------
+        9          | 19.738556
+        10         | 16.512998
+        11         | 22.946719
+
+        Returns:
+            [string] name of newly created table
+        """
+
+        self.drop_stale_feature_table("features.cell_timings")
+
+        query = ("CREATE TABLE features.cell_timings AS           "
+                 "  (SELECT exampleid, MAX(t_trace) -             "
+                 "  MIN(t_trace) as total_elapsed_time            "
+                 "  FROM raw.frontpage_traces GROUP BY exampleid);")
+        self.execute_query_from_string(query)
+
+        return "features.cell_timings"
+
+    def generate_table_interpacket_timings(self):
```

Comment:
It's honestly readable enough to pretty easily compare to Wang's spec at this point, and I do agree we should prioritize feature generation grouping according to what makes for good Python code rather than what make this like reading from the spec.

> If we get to the point of adding additional features, we might preserve this list of features by renaming the current main() function to generate_wang_feature_set() or something

Good idea.